airflow/www/extensions/init_views.py

```diff
@@ -73,8 +73,14 @@ def init_plugins(app):
     appbuilder = app.appbuilder
 
     for view in plugins_manager.flask_appbuilder_views:
-        log.debug("Adding view %s", view["name"])
-        appbuilder.add_view(view["view"], view["name"], category=view["category"])
+        try:
+            log.debug("Adding view %s with menu", view["name"])
+            appbuilder.add_view(view["view"], view["name"], category=view["category"])
+        except KeyError as e:
+            if e.args[0] == 'name':
```

Comment:
What is the expected behavior if the user does not provide the "view" key? It seems to me that in this case the exception will be silenced. It seems to me that in this case the exception will be silenced.

Reply:
Right, it should've re-raise exception in that case. Missed this one.

Reply:
Fixed.

Reply:
How about:
```py
name = view.get("name")
if name:
    log.debug("Adding view %s with menu", name)
    appbuilder.add_view(view["view"], name, category=view["category"])
else:
    # if 'name' key is missing, intent is to add view without menu
    log.debug("Adding view %s without menu", str(type(view["view"])))
    appbuilder.add_view_no_menu(view["view"])

```

Reply:
@turbaszek Yeah, that'd work too and I guess, it's better in terms of readability. I was just concerned about following python's EAFP principle here, but I guess it defeats the purpose if we're anyways checking for the "name" value. :thinking:

Reply:
@turbaszek implemented your suggestion :+1: